src/api/rate_limiter.py

```diff
@@ -0,0 +1,114 @@
+import asyncio
+import logging
+import uuid
+from datetime import datetime
+from os import getenv
+import requests
+
+
+
+class RequestEntry() :
+    """
+    Class for request object to put in the queue for the rate limiter
+    """
+    def __init__(self,url,cookies,key,method) -> None:
+        self.url = url
+        self.cookies = cookies
+        self.key = key
+        self.method = method
+
+class RateLimiter() :
+    """
+    Class that takes care of sending the requests at a maximum rate (25/sec) 
+        and giving back the result to the calling functions when it is received (async probably)
+    """
+    _max_retry = 0
+
+    def __init__(self) :
+        self.requests = {}
+        self.queue = asyncio.Queue()
+        #set a max_retry cap
+        if getenv("MAX_API_RETRY") is not None :
+            self._max_retry = int(getenv("MAX_API_RETRY"))
```

Comment:
Actually the static variables seemed fine to me as the `_max_retry` doesn't seems too instance-dependant, but this do the job as well!

Reply:
@Xlitoni if you wish to let it as is please yourself to resolve this thread yourself and merge the PR I'll approve it :wink: 
Otherwise you can change it to a static attribute that would be cool too!